File: leetcode/pattern/2-two-pointers/2-two-pointers.py
from typing import List 

class Solution:
  def two_pointers(self, nums: List[int], target: int) -> List[int]:
    left = 0
    right = len(nums)-1

    try:
      while True:
        if target == nums[left] + nums[right]:
          return [left, right]

        if target > nums[left] + nums[right]:
          left += 1
          continue 

        if target < nums[left] + nums[right]:
          right -= 1
          continue 
    except:
      return []


s = Solution()
print(
  [1, 3] == s.two_pointers([1, 2, 3, 4, 6], 6)
)

File: leetcode/pattern/2-two-pointers/2-two-pointers.go
package main

import (
	"fmt"
	"reflect"
)

func main() {
	got := twoPointers([]int{1, 2, 3, 4, 6}, 6)
	want := []int{1, 3}

	fmt.Println(got)
	fmt.Println(want)
	if !reflect.DeepEqual(got, want) {
		fmt.Println("test failed")
	}
}

func twoPointers(nums []int, target int) []int {
	left := 0
	right := len(nums) - 1

	for left < right {
		sum := nums[left] + nums[right]

		if sum == target {
			return []int{left, right}
		}

		if sum < target {
			left++
		} else {
			right--
		}
	}

	return []int{}
}
